packages/react-components/src/components/Dialpad/Dialpad.tsx

```diff
@@ -0,0 +1,222 @@
+  // bit of formatting and return it.
+  return `${countryCodeNA}(${phoneNumber.slice(0, 3)}) ${phoneNumber.slice(3, 6)}-${phoneNumber.slice(
+    6,
+    phoneNumber.length
+  )}`;
+};
+
+const DialpadButton = (props: {
+  primaryContent: string;
+  secondaryContent?: string;
+  styles?: Partial<DialpadStyles>;
+  onClick: (input: string) => void;
+}): JSX.Element => {
+  const theme = useTheme();
+  return (
+    <DefaultButton
+      onClick={() => {
+        props.onClick(props.primaryContent);
+      }}
+      styles={concatStyleSets(buttonStyles(theme), props.styles?.button)}
+    >
+      <Stack>
+        <Text className={mergeStyles(digitStyles(theme), props.styles?.digit)}>{props.primaryContent}</Text>
+
+        <Text className={mergeStyles(subStyles(theme), props.styles?.subDigit)}>{props.secondaryContent ?? ' '}</Text>
+      </Stack>
+    </DefaultButton>
+  );
+};
+
+const dialPadButtonsDefault: DialpadButtonsType[][] = [
+  [
+    { primaryContent: '1' },
+    { primaryContent: '2', secondaryContent: 'ABC' },
+    { primaryContent: '3', secondaryContent: 'DEF' }
+  ],
+  [
+    { primaryContent: '4', secondaryContent: 'GHI' },
+    { primaryContent: '5', secondaryContent: 'JKL' },
+    { primaryContent: '6', secondaryContent: 'MNO' }
+  ],
+  [
+    { primaryContent: '7', secondaryContent: 'PQRS' },
+    { primaryContent: '8', secondaryContent: 'TUV' },
+    { primaryContent: '9', secondaryContent: 'WXYZ' }
+  ],
+  [{ primaryContent: '*' }, { primaryContent: '0', secondaryContent: '+' }, { primaryContent: '#' }]
+];
+
+const DialpadContainer = (props: {
+  errorText: string;
+  defaultText: string;
+  dialpadButtons?: DialpadButtonsType[][];
+  styles?: Partial<DialpadStyles>;
+}): JSX.Element => {
+  const theme = useTheme();
+  const [textValue, setTextValue] = useState('');
+  const [error, setError] = useState('');
+  const onClickDialpad = (input: string): void => {
+    setError('');
+
+    setTextValue(textValue + input);
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const setText = (e: any): void => {
+    // if text content includes non-valid dialpad input return an error
+    if (!`${e.target.value}`.match(/^[0-9\s*#+()-]*$/)) {
+      setError(props.errorText);
+    } else {
+      setError('');
+      setTextValue(e.target.value);
+    }
+  };
+
+  const dialpadButtonsContent = props.dialpadButtons ?? dialPadButtonsDefault;
+
+  return (
+    <div className={mergeStyles(containerStyles(theme), props.styles?.root)}>
+      <TextField
+        styles={concatStyleSets(textFieldStyles(theme), props.styles?.textField)}
+        value={formatPhoneNumber(textValue)}
+        onChange={setText}
+        errorMessage={error}
+        placeholder={props.defaultText}
+      />
+
+      {dialpadButtonsContent.map((rows, i) => {
+        return (
+          <Stack horizontal key={`row_${i}`} style={{ alignItems: 'stretch' }}>
```

Comment:
can `style={{ alignItems: 'stretch' }}` be avoided and instead use Stack `horizontalAlign` preoprty or the StackItem `align` property directly

Reply:
updated